vars/jenkinsMaterializeLog.groovy

```diff
@@ -0,0 +1,80 @@
+import com.sap.piper.ConfigurationHelper
+import com.sap.piper.GenerateDocumentation
+import com.sap.piper.k8s.ContainerMap
+import hudson.FilePath
+import com.sap.piper.Utils
+import groovy.transform.Field
+import java.util.UUID
+import java.io.File
+import com.cloudbees.groovy.cps.NonCPS
+import com.sap.piper.JenkinsUtils
+
+import static com.sap.piper.Prerequisites.checkScript
+
+@Field String STEP_NAME = getClass().getName()
+@Field Set GENERAL_CONFIG_KEYS = []
+
+@Field Set STEP_CONFIG_KEYS = GENERAL_CONFIG_KEYS
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+
+/**
+ * This step allows you to materialize the Jenkins log file of the running build
+ */
+@GenerateDocumentation
+void call(Map parameters = [:], body) {
+	handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters)
+	{
+		def jenkinsUtils = parameters.jenkinsUtilsStub ?: new JenkinsUtils()
+		checkScript(this, parameters) ?: this
+		withMaterializedLogFile(body, jenkinsUtils)
+	}
+}
+
+
+@NonCPS
+def writeLogToFile(fp) {
+	def logInputStream = currentBuild.rawBuild.getLogInputStream()
+	fp.copyFrom(logInputStream)
+	logInputStream.close()

```

Comment:
Is the `copyFrom` above executed synchronously? Is it fine to close the steam here. Should it closed by some kind of framework at end of build. Does somebody else need that input stream? Maybe everything is fine here, I don't have the domain knowledge to judge. Somebody else should cross check IMO: